pyclam/manifold.py

```diff
@@ -715,24 +716,26 @@ def _partition_threaded(self, criterion):
 
         return clusters
 
-    def select(self, name: str) -> Cluster:
-        """ Returns the cluster with the given name. """
+    def ancestry(self, name: str) -> List[Cluster]:
         if len(name) > self.depth:
             raise ValueError(f'depth of requested cluster must not be greater than depth of cluster-tree. Got {name}, max-depth: {self.depth}')
 
         # TODO: Consider how to change this for forests.
-        cluster: Cluster = next(iter(self.graphs[0]))
+        root: Cluster = next(iter(self.graphs[0]))
+        lineage = [root]
         for depth in range(len(name) + 1):
-            partial_name = name[:depth]
-            for child in cluster.children:
-                if child.name == partial_name:
-                    cluster = child
+            for child in lineage[-1].children:
+                if child.name == name[:depth]:
```

Comment:
Should we make children of a cluster a set keyed on name?

Reply:
There are only one or two children so I do not see the change being worth it